app/src/main/java/com/example/music_app/ui/screens/new_playlist/NewPlaylistScreen.kt

```diff
@@ -26,40 +30,55 @@ fun NewPlaylistScreen(
     onCreatePlaylistSuccess: () -> Unit
 ) {
 
+    val error by viewModel.error.collectAsStateWithLifecycle()
     val nameFieldState by viewModel.nameFieldState.collectAsStateWithLifecycle()
     val descriptionFieldState by viewModel.descriptionFieldState.collectAsStateWithLifecycle()
     val isNameValid by viewModel.isNameValid.collectAsStateWithLifecycle()
     val isFormValid by viewModel.isFormValid.collectAsStateWithLifecycle()
     val isNameFieldActive by viewModel.isNameFieldActive.collectAsStateWithLifecycle()
 
-    Column(
-        modifier = Modifier
-            .fillMaxSize()
-            .background(MaterialTheme.colorScheme.background)
-            .padding(AppTheme.dimens.spacing10),
-        verticalArrangement = Arrangement.spacedBy(AppTheme.dimens.spacing10),
-        horizontalAlignment = CenterHorizontally
-    ) {
-        NewPlaylistInputsField(
-            nameFieldState = nameFieldState,
-            descriptionFieldState = descriptionFieldState,
-            isNameValid = isNameValid,
-            isNameFieldActive = isNameFieldActive,
-            modifier = Modifier
-                .fillMaxWidth()
-                .weight(1f),
-            onNameValueChange = viewModel::onNameValueChange,
-            onDescriptionValueChange = viewModel::onDescriptionValueChange
-        )
-        ButtonField(
-            text = stringResource(R.string.create_button),
-            onClick = {
-                viewModel.onCreatePlaylist(onCreatePlaylistSuccess)
-            },
+    val snackbarHostState = remember { SnackbarHostState() }
+
+    ErrorLaunchedEffect(
+        error = error,
+        snackbarHostState = snackbarHostState
+    )
+
+    ErrorSnackbar(
+        errorId = error?.errorId ?: R.string.error,
+        snackbarHostState = snackbarHostState,
+        onClick = { viewModel.onError() }
+    ) { paddingValues ->
```

Comment:
`Applies to all occurrences` Could this component be a parent to all screens that supports error handling to prevent code duplication ?